scripts/Update-OnETagChanged.ps1

```diff
@@ -27,24 +29,52 @@ function Update-OnETagChanged() {
     $saveResult = $true
   }
   else {
-    $existingInfo = (Get-Content $saveFile -Encoding UTF8 -TotalCount 1) -split '\|'
-
-    if ($existingInfo[0] -ne $etag) {
-      $result = . $OnETagChanged
-      $saveResult = $true
-    }
-    else {
-      $result = . $OnUpdated
-      $result["Version"] = $existingInfo[1]
-      $result["ETAG"] = $existingInfo[0]
-      $saveResult = $false
-    }
+	  if ($kind) {
+		$existingInfo = ( Get-Content $saveFile -Raw | ConvertFrom-Json )
+    # Conversion of stored etag to have double quotes like web ETag
+        $existingETag = $existingInfo.$kind -replace("'",'"')
+		if ($existingETag -ne $etag) {
+		  $result = . $OnETagChanged
+		  $saveResult = $true
+		}
+		else {
+		  $result = . $OnUpdated
+		  $result["ETAG"] = $existingETag
+		  $saveResult = $false
+		}
+        if ($saveResult) {
+        # Conversion of ETag to have single quotes
+        $etag = $etag -replace('"', "'")
+            foreach($line in $existingInfo){
+	            $existingInfo.$kind = $line.$kind -replace("([W]\/)?('((\d+)?(\w)?(\-)?)+')" , $etag)
+            }
+        $result = $etag
+        $existingInfo | ConvertTo-Json -depth 32 | Set-Content $saveFile
+        }
+	  }
+	  else {
+	    $existingInfo = (Get-Content $saveFile -Encoding UTF8 -TotalCount 1) -split '\|'
+		if ($existingInfo[0] -ne $etag) {
+		  $result = . $OnETagChanged
+		  $saveResult = $true
+		}
+		else {
+		  $result = . $OnUpdated
+		  $result["Version"] = $existingInfo[1]
+		  $result["ETAG"] = $existingInfo[0]
+		  $saveResult = $false
+		}
+        if ($saveResult) {
+        $result["ETAG"] = $etag
+        "$($result["ETAG"])|$($result["Version"])" | Out-File $saveFile -Encoding utf8 -NoNewline
+        }
+	  }
+Write-Warning "existingETag -$existingETag-`r`netag -$etag-"
   }
-
-  if ($saveResult) {
-    $result["ETAG"] = $etag
-    "$($result["ETAG"])|$($result["Version"])" | Out-File $saveFile -Encoding utf8 -NoNewline
+  if ($boule) {
+	return $saveResult
```

Comment:
By just returning a `$true` or `$false` value here, you are changing the intention of the script.

The return value from the helper should be compatible with what AU expects, in this case, a hash table containing the information necessary.